src/js/lib/helpers.js

```diff
@@ -150,6 +153,28 @@
 		return decodeURIComponent(match[1].replace(/\+/g, ' '));
 	};
 
+	/*
+	 * Find dynamic context generating functions and merge their results into the static contexts
+	 * Combine an array of unchanging contexts with the result of a context-creating function
+	 *
+	 * @param {(object|function(...*): ?object)[]} dynamicOrStaticContexts Array of custom context Objects or custom context generating functions
+	 * @param {...*} Parameters to pass to dynamic callbacks
+	 */
+	object.resolveDynamicContexts = function (dynamicOrStaticContexts) {
+		let params = Array.prototype.slice.call(arguments, 1);
+		return map(dynamicOrStaticContexts, function(context) {
+			if (typeof context === 'function') {
+				try {
+					return context.apply(null, params);
+				} catch (e) {
+					//TODO: provide warning
```

Comment:
Should this TODO be addressed?

Reply:
I was going to save this for when there's a proper debug mode implemented - we avoid outputting to the console.